services/catalog/src/simcore_service_catalog/api/routes/services.py

```diff
@@ -0,0 +1,258 @@
+import logging
+import urllib.parse
+from typing import List, Set, Tuple
+
+from fastapi import APIRouter, Depends, HTTPException, status
+from pydantic import ValidationError, constr
+from pydantic.types import PositiveInt
+
+from ...db.repositories.groups import GroupsRepository
+from ...db.repositories.services import ServicesRepository
+from ...models.domain.service import (
+    KEY_RE,
+    VERSION_RE,
+    ServiceAccessRightsAtDB,
+    ServiceMetaDataAtDB,
+    ServiceUpdate,
+    ServiceOut,
+)
+from ..dependencies.database import get_repository
+from ..dependencies.director import AuthSession, get_director_session
+
+router = APIRouter()
+logger = logging.getLogger(__name__)
+
+# FIXME: too many DB calls
+
+
+@router.get("", response_model=List[ServiceOut])
+async def list_services(
+    user_id: PositiveInt,
+    director_client: AuthSession = Depends(get_director_session),
+    groups_repository: GroupsRepository = Depends(get_repository(GroupsRepository)),
+    services_repo: ServicesRepository = Depends(get_repository(ServicesRepository)),
+):
+    # get user groups
+    user_groups = await groups_repository.list_user_groups(user_id)
+    if not user_groups:
+        # deny access
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="You have unsufficient rights to access the services",
+        )
+    # now get the executable services
+    executable_services: Set[Tuple[str, str]] = {
+        (service.key, service.version)
+        for service in await services_repo.list_services(
+            gids=[group.gid for group in user_groups], execute_access=True
+        )
+    }
+    # get the writable services
+    writable_services: Set[Tuple[str, str]] = {
+        (service.key, service.version)
+        for service in await services_repo.list_services(
+            gids=[group.gid for group in user_groups], write_access=True
+        )
+    }
+    # get the services from the registry and filter them out
+    data = await director_client.get("/services")
```

Comment:
instead of a direct all, why don't you rely on the data collected by the background task ``core/backgroud_tasks.py:sync_registry_task`` ??

Reply:
I don't copy everything in the DB and do still rely on the director for this. I thought I might do that in a second iteration maybe.

Reply:
and the PR is already quite big...